src/lib.rs

```diff
@@ -105,101 +102,93 @@ impl Plugin for BevyLocalCommandsPlugin {
     }
 }
 
-fn handle_new_process(
-    mut run_process_event: EventReader<RunProcess>,
-    mut process_started_event: EventWriter<ProcessStarted>,
+/// A new command has been added.
+///
+/// This system will spawn the corresponding process and track the process output.
+fn handle_new_command(
+    mut commands: Commands,
+    mut query: Query<(Entity, &mut LocalCommand), (Added<LocalCommand>, Without<Process>)>,
     mut process_error_event: EventWriter<ProcessError>,
-    mut active_process_map: ResMut<ActiveProcessMap>,
 ) {
-    for run_shell_command in run_process_event.read() {
-        // Assemble the command
-        let mut cmd = Command::new(run_shell_command.program.clone());
-        cmd.args(run_shell_command.arguments.clone())
-            .stdout(Stdio::piped());
-
-        let command = format!("{cmd:?}");
-
-        let Ok(active_process) = spawn_process(cmd) else {
-            process_error_event.send(ProcessError {
-                command,
-                info: ProcessErrorInfo::FailedToStart,
-            });
-            continue;
-        };
-
-        let pid = active_process.process.id();
-
-        process_started_event.send(ProcessStarted { command, pid });
-
-        active_process_map.0.insert(pid, active_process);
+    for (entity, mut local_command) in query.iter_mut() {
+        match spawn_process(&mut local_command.command) {
+            Ok(process) => {
+                commands.entity(entity).insert(process);
+            },
+            Err(_) => {
+                process_error_event.send(ProcessError {
```

Comment:
If we're despawning entities when the process completes, do we want to do the same when the process fails?

Reply:
Hmm, good question.

The automatic despawning is one area that I'm unsure about, as the user could also have attached additional information to the entity in different components.
But maybe that's something we can add as a configuration option later.
Or we add a marker component which prevents automatic despawning, similar to the other add on ideas.
Then the user has fine-grained control if they need it.